index.js

```diff
@@ -79,13 +162,119 @@ async function iterateOverCollectionDefinitionAndRegisterCruds(fastify) {
     fastify.register(registerCrud, {
       modelName,
-      isView: model.isView,
+      isView,
     })
+
+    if (viewLookupsEnabled) {
+      const lookups = viewDependencies.lookupsModels.map(({ lookup }) => lookup)
+      fastify.register(registerViewCrud, {
+        modelName,
+        lookups,
+      })
+
+      for (const lookupModel of viewDependencies.lookupsModels) {
+        fastify.register(registerViewCrudLookup, {
+          modelName,
+          lookupModel,
+        })
+      }
+    }
   }
 }
 
+function buildModelDependencies(fastify, collection, collectionDefinition) {
+  const {
+    schema: collectionSchema,
+    fields: deprecatedCollectionSchema,
+    defaultState,
+  } = collectionDefinition
+
+  const allFieldNames = !collectionSchema
+    ? deprecatedCollectionSchema.map(({ name }) => name)
+    : Object.keys(collectionDefinition.schema.properties)
+  const pathsForRawSchema = generatePathFieldsForRawSchema(fastify.log, collectionDefinition)
+
+  // TODO: make this configurable
+  const crudService = new CrudService(
+    collection,
+    defaultState,
+    { allowDiskUse: fastify.config.ALLOW_DISK_USE_IN_QUERIES },
+  )
+  const queryParser = new QueryParser(collectionDefinition, pathsForRawSchema)
+  const resultCaster = new ResultCaster(collectionDefinition)
+  const jsonSchemaGenerator = new JSONSchemaGenerator(
+    collectionDefinition,
+    {},
+    fastify.config.CRUD_LIMIT_CONSTRAINT_ENABLED,
+    fastify.config.CRUD_MAX_LIMIT
+  )
+  const jsonSchemaGeneratorWithNested = new JSONSchemaGenerator(
+    collectionDefinition,
+    pathsForRawSchema,
+    fastify.config.CRUD_LIMIT_CONSTRAINT_ENABLED,
+    fastify.config.CRUD_MAX_LIMIT
+  )
+
+  return {
+    crudService,
+    queryParser,
+    resultCaster,
+    allFieldNames,
+    jsonSchemaGenerator,
+    jsonSchemaGeneratorWithNested,
+    castResultsAsStream: () => resultCaster.asStream(),
+    castItem: (item) => resultCaster.castItem(item),
+  }
+}
+
+function createLookupModel(fastify, viewDefinition, mergedCollections) {
+  const lookupModels = []
+  const viewLookups = viewDefinition.pipeline
+    .filter(pipeline => '$lookup' in pipeline)
+    .map(lookup => Object.values(lookup).shift())
+
+  for (const lookup of viewLookups) {
+    const { from, pipeline } = lookup
+    const lookupCollection = clone(mergedCollections.find(({ name }) => name === from))
+    const lookupIdType = getIdType(lookupCollection)
+    const lookupCollectionMongo = fastify.mongo[getDatabaseNameByType(lookupIdType)].db.collection(from)
+
+    const lookupProjection = pipeline.find(({ $project }) => $project !== undefined)?.$project ?? {}
+    const parsedLookupProjection = []
+    const lookupCollectionDefinition = {
+      ...omit(['fields'], viewDefinition),
+      schema: {
+        type: 'object',
+        properties: {},
+        required: [],
+      },
+    }
+
+    Object.entries(lookupProjection)
+      .forEach(([key, value]) => {
```

Comment:
Is it possible to assign less generic names to `key` and `value` variables (something that explains their meaning?)